sdc/rewrites/dataframe_getitem_attribute.py

```diff
@@ -0,0 +1,97 @@
+# *****************************************************************************
+# Copyright (c) 2020, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+from numba.ir import Assign, Const, Expr, Var
+from numba.ir_utils import mk_unique_var
+from numba.rewrites import register_rewrite, Rewrite
+from numba.types import StringLiteral
+from numba.typing import signature
+
+from sdc.config import config_pipeline_hpat_default
+from sdc.hiframes.pd_dataframe_type import DataFrameType
+
+
+if not config_pipeline_hpat_default:
+    @register_rewrite('after-inference')
+    class RewriteDataFrameGetItemAttr(Rewrite):
+        """
+        Search for calls of df.attr and replace it with calls of df['attr']:
+        $0.2 = getattr(value=df, attr=A) -> $const0.0 = const(str, A)
+                                            $0.2 = static_getitem(value=df, index=A, index_var=$const0.0)
+        """
+
+        def match(self, func_ir, block, typemap, calltypes):
+            self.func_ir = func_ir
+            self.block = block
+            self.typemap = typemap
+            self.calltypes = calltypes
+            self.getattrs = getattrs = set()
+            for expr in block.find_exprs(op='getattr'):
+                obj = typemap[expr.value.name]
+                if not isinstance(obj, DataFrameType):
+                    continue
+                if expr.attr in obj.columns:
+                    getattrs.add(expr)
```

Comment:
So we won't rewrite if we didn't find valid column name, right?
This means we won't raise AttributeError as pandas does if we use an invalid attribute name. We should probably state that in limitations, or think if we can look for all_supported_dataframe_methods in this rewrite and if we actually call a method skip rewriting, which aligns to pandas behavior:
```
>>>df
   A  B  values
0  2 -8      -8
1  1  2       2
2  1  3       3
3  1  1       1
4  2  5       5
5  2  6       6
6  1  7       7
>>>df.values
array([[ 2, -8, -8],
       [ 1,  2,  2],
       [ 1,  3,  3],
       [ 1,  1,  1],
       [ 2,  5,  5],
       [ 2,  6,  6],
       [ 1,  7,  7]], dtype=int64)
```

Reply:
Yes, we need to think about it, but in next PR

Reply:
Right now, the main issue us the case when column is named like '_data', '_columns' or '_index'. This will break all uses of dataframe